### PR TITLE
Validating an existing file

### PR DESCRIPTION
**Pull request by msuchane**
Resolves issue #11.

### CODE REVIEW

File: src/validation.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    }
+}
+
+fn determine_mod_type(base_name: &str, content: &str) -> Option<ModuleType> {
+    // An inner function to encapsulate the logic.
+    // This can't be a simple expression block because the logic involves
+    // stuff like for loops.
+    let inner = || {
+        if base_name.starts_with("assembly_") || base_name.starts_with("assembly-") {
+            return Some(ModuleType::Assembly);
```
Comment (VladimirSlavik) [inline src/validation.rs]: Are all assemblies called `assembly[_-].*` ?

Comment (msuchane) [reply src/validation.rs]: Well, some projects use modules with no prefixes, and that probably applies to assemblies as well. So there can conceivably be assembly files without the `assembly[_-].*` prefix.

The way that the validation detects the module type needs to be much more sophisticated than it is right now, anyway. It has to deal with cases where a file looks like one type in some places but another type in other places, and report that as an error.

For assemblies, my current idea is that if a file has no prefix and it includes other modules, `newdoc` should report "Please check if this is really an assembly".

Comment (VladimirSlavik) [reply src/validation.rs]: Ah, so it's a first version? I'm so used to draft PRs that I thought this is "finished"...

Comment (msuchane) [reply src/validation.rs]: Absolutely, this is a work in progress, an early one. See the "WIP" in the title :-)

I'll let you know when it's finished, if you'd like to take a look then.

**Comment by msuchane** [comment]
I believe the validation feature now covers the whole Pv2 checklist. I'm removing the WIP label to reflect that.

As the next step, I'll do some more testing, and I'll probably reorganize the code to make it more readable.

**Comment by msuchane** [comment]
@VladimirSlavik, the validation feature is now pretty much complete and ready for a review. Are you still interested? :-)

If you are, I really have no expectations on how thorough (or cursory) the review would be, so I leave it to you and your time schedule.

**Comment by VladimirSlavik** [comment]
I will try to take a look. At 60-ish commits, it might be hard to actually get it all :)

**Comment by VladimirSlavik** [comment]
I think this is fine. Thanks for the opportunity to review!

**Comment by msuchane** [comment]
Thanks for having a look, @VladimirSlavik! Now I can consider this to be reviewed by a pro.

**Comment by VladimirSlavik** [comment]
Please don't - rust is hardly something I *know*.

**Comment by msuchane** [comment]
@VladimirSlavik, noted, but I still appreciate your help.

This branch is now merged and I'm releasing the new version to the writers.

**Pull request merged**